src/lib/plugins/live-cell/live-cell.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input, OnInit, SimpleChange, ChangeDetectionStrategy, NgZone } from '@angular/core';
+import { TemplateHostService } from '../../template/template-host.service';
+import { TdCoreDirective } from 'lib/main/core/body/td-core.directive';
+import { AppError } from '../../../core/infrastructure/error';
+
+@Component({
+	selector: 'q-grid-live-cell',
+	templateUrl: './live-cell.tpl.html',
+	providers: [TemplateHostService],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LiveCellComponent implements OnInit {
+	@Input() cell: TdCoreDirective;
+	@Input() duration = 500;
+
+	changes: any;
+
+	constructor(private zone: NgZone) {
+	}
+
+	ngOnInit() {
+		if (!this.cell.changes) {
+			throw new AppError('q-grid-live-cell', 'Changes is not defined in live-cell.component.ts');
+		}
+
+		this.changes = this.getDifference(this.cell.changes);
+		this.zone.runOutsideAngular(() => {
+			setTimeout(() => {
+				this.cell.mode('view');
+			}, this.duration);
+		});
+	}
+
+	getDifference(value: SimpleChange) {
```

Comment:
we need to use different method for numbers strings date etc.

Reply:
yep, so that's why i use switch-case in this method, depends on type of variable. Is it wrong?
we can get difference, when we working with numbers. In case of strings (dates, and so on) we just get previous value. Or how should i determine the type of input data? For example, for dates it's just a number. And how we should calculate difference for objects (arrays), Boolean, etc.

Reply:
Let's not persist it, and propagate to the template `getDiff()` method, we can apply cache in future if face with performance issues.